### PR TITLE
Move relstorage install to builder

### PR DESCRIPTION
**Pull request by sneridagh**
Since it breaks in M1/ARM64 builds, because gcc is not present in the later base build.

Did we have any reason for it to be installed later?

### CODE REVIEW

**Review by avoinea — DISMISSED** [review]

**Comment by jensens** [comment]
"In the case of psycopg2, that binary package (which is not recommended for production use) can be installed with the name psycopg2-binary. Note that the postgresql extra in RelStorage does not install the binary but attempts to install from source."
https://relstorage.readthedocs.io/en/latest/install.html#database-adapter

Do we use the binary already?

**Comment by jensens** [comment]
Maybe using the binary package is not that a good idea 
https://github.com/zodb/relstorage/issues/488

**Comment by sneridagh** [comment]
@jensens what should be the course of action then? Not use the binary?

Also, what do you think about moving them to the build step?

File: 5.2/5.2.6/Dockerfile.python36 [hunk 1]
```diff
@@ -4,6 +4,7 @@ FROM base as builder
 ENV PIP_PARAMS="--use-deprecated legacy-resolver"
 ENV PLONE_VERSION=5.2.6
 ENV PLONE_VOLTO="plone.volto==3.1.0a2"
+ENV EXTRA_PACKAGES="relstorage==3.4.5 psycopg2-binary==2.9.1"
```
Comment (jensens) [inline 5.2/5.2.6/Dockerfile.python36:7]:
```suggestion
ENV EXTRA_PACKAGES="relstorage==3.4.5 psycopg2==2.9.3"
```

Comment (jensens) [reply 5.2/5.2.6/Dockerfile.python36:7]: and so on in the other files

**Comment by jensens** [comment]
We must not use the psycopg2-binary in our official image, if it is not recommended. Unless psycopg2 is not distributed officially as binary wheels I would build them ourselfes.

But wait, `pip wheel` does build the wheel with gcc and in the target arch, right? So it's the same as `setup.py bdist_wheel`. And there we have the gcc available. No reason to use the binary package at all, and your PR is fine with just `psycopg2`. Or what do I miss?

And attention: The build step may run on outdated libraries, there no `apt-get update`/`apt-get upgrade` in there.

**Comment by jensens** [comment]
Probably psycopg 3 will solve all our problems at some point https://www.psycopg.org/articles/2021/10/13/psycopg-30-released/

**Comment by sneridagh** [comment]
@jensens ok, will try! Also include an update step in build.

**Comment by sneridagh** [comment]
@jensens green now. I tested it also locally with postgres and works well.

**Review by jensens — APPROVED** [review]

**Review by avoinea — APPROVED** [review]

**Pull request merged**